adafruit_cursorcontrol/cursorcontrol_cursormanager.py

```diff
@@ -0,0 +1,163 @@
+`adafruit_cursorcontrol_cursormanager`
+================================================================================
+Simple interaction user interface interaction for Adafruit_CursorControl.
+* Author(s): Brent Rubell
+"""
+import board
+import digitalio
+from micropython import const
+import analogio
+from gamepadshift import GamePadShift
+
+# PyBadge
+PYBADGE_BUTTON_LEFT = const(128)
+PYBADGE_BUTTON_UP = const(64)
+PYBADGE_BUTTON_DOWN = const(32)
+PYBADGE_BUTTON_RIGHT = const(16)
+# PyBadge & PyGamer
+PYBADGE_BUTTON_A = const(2)
+
+JOY_X_CTR = 32767.5
+JOY_Y_CTR = 32767.5
+
+class CursorManager:
+    """Simple interaction user interface interaction for Adafruit_CursorControl.
+
+    :param adafruit_cursorcontrol cursor: The cursor object we are using.
+    """
+    def __init__(self, cursor):
+        self._cursor = cursor
+        self._is_clicked = False
+        self._init_hardware()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.deinit()
+
+    def deinit(self):
+        """Deinitializes a CursorManager object."""
+        self._is_deinited()
+        self._pad.deinit()
+        self._cursor.deinit()
+        self._cursor = None
+
+    def _is_deinited(self):
+        """Checks if CursorManager object has been deinitd."""
+        if self._cursor is None:
+            raise ValueError("CursorManager object has been deinitialized and can no longer "
+                             "be used. Create a new CursorManager object.")
+
+    def _init_hardware(self):
+        """Initializes PyBadge or PyGamer hardware."""
+        if hasattr(board, 'BUTTON_CLOCK') and not hasattr(board, 'JOYSTICK_X'):
+            self._pad_btns = {'btn_left' : PYBADGE_BUTTON_LEFT,
+                              'btn_right' : PYBADGE_BUTTON_RIGHT,
+                              'btn_up' : PYBADGE_BUTTON_UP,
+                              'btn_down' : PYBADGE_BUTTON_DOWN,
+                              'btn_a' : PYBADGE_BUTTON_A}
+        elif hasattr(board, 'JOYSTICK_X'):
+            self._joystick_x = analogio.AnalogIn(board.JOYSTICK_X)
+            self._joystick_y = analogio.AnalogIn(board.JOYSTICK_Y)
+            self._pad_btns = {'btn_a' : PYBADGE_BUTTON_A}
+        else:
+            raise AttributeError('Board must have a D-Pad or Joystick for use with CursorManager!')
+        self._pad = GamePadShift(digitalio.DigitalInOut(board.BUTTON_CLOCK),
+                                 digitalio.DigitalInOut(board.BUTTON_OUT),
+                                 digitalio.DigitalInOut(board.BUTTON_LATCH))
+
+    @property
+    def is_clicked(self):
+        """Returns True if the cursor button was pressed
+        during previous call to update()
+        """
+        return self._is_clicked
+
+    def update(self):
+        """Updates the cursor object."""
+        pressed = self._pad.get_pressed()
+        self._check_cursor_movement(pressed)
+        if self._is_clicked:
```

Comment:
check this logic here, which could be where your extra click is comin' from! otherwise library looks good :)

Reply:
Updated in latest!